### ISSUE TITLE
backend_generic.cpp: lm_head() always returns false — breaks cascade/adaptive strategy routing

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#958]
## Summary

In `src/backend_generic.cpp`, `lm_head()` is not implemented — it always returns `false`:

```cpp
bool lm_head(const float* hidden, float* logits, int* argmax) override {
    return false;  // not implemented — use generate() instead
}
```

This breaks the cascade and adaptive strategy engines in `unified_server.cpp`, which call `mgr.forward() + mgr.lm_head()` for per-token logprobs:

```cpp
if (mgr.lm_head(hidden_buf.data(), logits_buf.data(), &next)) {
    // compute logprob...
}
```

Since `lm_head()` always returns `false`, the cascade/adaptive strategies get no logprob feedback from the generic backend — they always fall back to the `-20.0` default logprob. This defeats the purpose of cascade routing.

## Suggested fix
Implement `lm_head()` for the generic backend using the same logic as the LM head in `forward()`:

```cpp
bool lm_head(const float* hidden, float* logits, int* argmax) override {
    const float* lm_w = output_weight.empty() ? embed.data() : output_weight.data();
    matmul(logits, hidden, lm_w, cfg.vocab, cfg.hidden);
    if (argmax) {
        *argmax = 0;
        for (int i = 1; i < cfg.vocab; ++i) {
            if (logits[i] > logits[*argmax]) *argmax = i;
        }
    }
    return true;
}
```

**Comment by bong-water-water-bong** [comment]
Fixed in PR #961 — lm_head() now properly computes logits via matmul instead of returning false

**Comment by bong-water-water-bong** [comment]
Fixed in PRs #951-#953, #961-#962, #969-#972
